### PR TITLE
Update APIv3 documentation with latest changes

### PR DESCRIPTION
**Pull request by humitos**
- Added Redirects documentation
- Remove Users documentation
- Added Project Import documentation
- Make sure the fields in the responses match the returned ones
- Add a note inviting users to request for beta testing

### CODE REVIEW

File: docs/api/v3.rst [hunk 1]
```diff
@@ -472,140 +522,230 @@ Build triggering
...
-Subprojects
-~~~~~~~~~~~
+Redirects
+~~~~~~~~~
 
-Subprojects listing
-+++++++++++++++++++
+Redirects allow the author to redirect an old URL of the documentation to a new one.
+This is useful when pages are moved around in the structure of the documentation set.
+See :doc:`user-defined-redirects` for more information.
```
Comment (humitos) [inline docs/api/v3.rst]: @stsewd do you know why this does not work? We are doing exactly the same in the `index.rst` page.

Comment (stsewd) [reply docs/api/v3.rst]: The doc directive search for relative paths, here you need to search from an absolute path (``:doc:`/user-defined-redirects``

Comment (humitos) [reply docs/api/v3.rst]: Thanks!

File: docs/api/v3.rst [hunk 2]
```diff
@@ -143,29 +154,29 @@ Project details
                 "virtualenv"
             ],
             "users": [
-                "dstufft",
-                "pmoore",
-                "xafer",
-                "pradyunsg"
+                {
+                    "created": "2010-10-23T18:12:31+00:00",
```
Comment (ericholscher) [inline docs/api/v3.rst]: I don't think we can to expose this data do we?

File: docs/api/v3.rst [hunk 3]
```diff
@@ -175,6 +186,47 @@ Project details
...
+
+    **Example request**:
+
+    .. sourcecode:: bash
+
+        $ curl \
+          -X POST \
+          -H "Authorization: Token <token>" https://readthedocs.org/api/v3/projects/ \
+          -H "Content-Type: application/json" \
+          -d @body.json
```
Comment (ericholscher) [inline docs/api/v3.rst]: Should we also include a Python example?

File: docs/api/v3.rst [hunk 4]
```diff
@@ -472,140 +522,230 @@ Build triggering
...
-Subprojects
-~~~~~~~~~~~
+Redirects
+~~~~~~~~~
 
-Subprojects listing
-+++++++++++++++++++
+Redirects allow the author to redirect an old URL of the documentation to a new one.
+This is useful when pages are moved around in the structure of the documentation set.
+See :doc:`/user-defined-redirects` for more information.
```
Comment (ericholscher) [inline docs/api/v3.rst]: We should probably put a similar description for all the resources (eg. Translations)

File: docs/api/v3.rst [hunk 5]
```diff
@@ -472,140 +522,230 @@ Build triggering
...
+
+    `See Redirect details <#redirect-details>`_
 
     :requestheader Authorization: token to authenticate.
+
+    :statuscode 200: Success
+    :statuscode 400: Some field is invalid
+
+
+Redirect destroy
```
Comment (ericholscher) [inline docs/api/v3.rst]: I think `delete` is probably more common than `destroy` -- we should standardize on one.

Comment (humitos) [reply docs/api/v3.rst]: He, yeah. The verb is DELETE but DRF uses destroy so :man_shrugging: 

I think delete makes more sense here.

File: docs/api/v3.rst (hunk 2 above)
Comment (humitos) [reply docs/api/v3.rst]: I'm removing it for now. We can add user's fields later if we consider without breaking anything.

File: docs/api/v3.rst (hunk 3 above)
Comment (humitos) [reply docs/api/v3.rst]: That would be pretty handy. I will create an issue for this to use tabs (Python + Slumber and cURL).

Comment (humitos) [reply docs/api/v3.rst]: Issue at https://github.com/readthedocs/readthedocs.org/issues/5935

**Comment by humitos** [comment]
@ericholscher I made some changes to the PR. Feel free to merge if you consider that's ready. Otherwise, let me know.

**Pull request merged**